### PR TITLE
Bump rust-sdk version to rust-sdk 0.2.57

### PR DESCRIPTION
**Pull request by BillCarsonFr**
## Content

Base PR to update all breaking changes with the new rust-sdk version. Do not merge yet.
Here is the stak of PR to be merged on this one:
- https://github.com/element-hq/element-x-android/pull/3724
- https://github.com/element-hq/element-x-android/pull/3729
- https://github.com/element-hq/element-x-android/pull/3736

## Tests

- Step 1
- Step 2
- Step ...

## Tested devices

- [ ] Physical
- [ ] Emulator
- OS version(s):

## Checklist

- [ ] Changes have been tested on an Android device or Android emulator with API 23
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#accessibility
- [ ] Pull request is based on the develop branch
- [ ] Pull request title will be used in the release note, it clearly define what will change for the user
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [ ] You've made a self review of your PR

### CODE REVIEW

File: tests/konsist/src/test/kotlin/io/element/android/tests/konsist/KonsistPreviewTest.kt [hunk 1]
```diff
@@ -119,6 +119,7 @@ class KonsistPreviewTest {
                 "TimelineItemGroupedEventsRowContentCollapsePreview",
                 "TimelineItemGroupedEventsRowContentExpandedPreview",
                 "TimelineItemImageViewHideMediaContentPreview",
+                "TimelineItemEventRowUtdPreview",
```
[the remark this answers is not in the document]
Comment (jmartinesp) [reply tests/konsist/src/test/kotlin/io/element/android/tests/konsist/KonsistPreviewTest.kt:122]: Ouch, I though I had placed it in the right place.

File: libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/timeline/item/event/EventTimelineItemMapper.kt [hunk 2]
```diff
@@ -78,25 +78,33 @@ fun RustEventSendState?.map(): LocalEventSendState? {
...
-                LocalEventSendState.Sending
-            } else {
-                LocalEventSendState.Failed.Unknown(error)
+            when (error) {
+                QueueWedgeError.CrossVerificationRequired -> {
+                    // The current device is not cross-signed (or cross signing is not setup)
+                    LocalEventSendState.Failed.SendingFromUnverifiedDevice
+                }
+                is QueueWedgeError.IdentityViolations -> {
+                    // smart cast impossible because 'error' is a public API property declared in different module
```
Comment (bmarty) [inline libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/timeline/item/event/EventTimelineItemMapper.kt:87]: Once the screenshot will be recorder, I'll push a commit to fix this.

**Review by bmarty — APPROVED** [review]
LGTM, thanks!

**Comment by jmartinesp** [comment]
Ah, my bad, I re-ran Maestro when it had already succeeded before.

**Pull request merged**